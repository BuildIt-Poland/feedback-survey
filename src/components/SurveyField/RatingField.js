import React from 'react';
import PropTypes from 'prop-types';

import { formField, radioOption } from '../../types';
import CustomRadioButtonGroup from '../CustomRadioButtonGroup';
import StyledFormField from './StyledFormField';

const RatingField = ({ field, options }) => (
  <StyledFormField
    name={field.id}
    label={field.content}
    required={field.required}
    component={CustomRadioButtonGroup}
    options={options}
    direction="row-responsive"
    width="48rem"
    withIcon={true}
  />
);

RatingField.propTypes = {
  field: formField.isRequired,
  options: PropTypes.arrayOf(radioOption)
};

export default RatingField;
