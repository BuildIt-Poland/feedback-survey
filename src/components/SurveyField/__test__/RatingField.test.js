import React from 'react';
import { create } from 'react-test-renderer';

import RatingField from '../RatingField';

describe('COMPONENT - SurveyField RatingField', () => {
  it('renders correctly', () => {
    const component = create(<RatingField />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
