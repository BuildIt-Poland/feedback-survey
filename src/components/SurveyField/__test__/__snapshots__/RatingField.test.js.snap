// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyField RatingField renders correctly 1`] = `null`;
